crates/agglayer-jsonrpc-api/src/admin.rs

```diff
@@ -27,8 +30,43 @@ pub enum ProcessNow {
     False,
 }
 
+use serde_with::{serde_as, DisplayFromStr};
+
+/// Token balance entry structure in order to display the balance tree values.
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TokenBalanceEntry {
+    pub origin_network: NetworkId,
+    pub origin_token_address: Address,
+    #[serde_as(as = "DisplayFromStr")]
+    pub amount: U256,
+}
+
+impl From<(SmtPath<192>, Digest)> for TokenBalanceEntry {
+    fn from((path, leaf_value): (SmtPath<192>, Digest)) -> Self {
+        let TokenInfo {
+            origin_network,
+            origin_token_address,
+        } = TokenInfo::from_bits(&path.as_bits());
+
+        Self {
+            origin_network,
+            origin_token_address,
+            amount: U256::from_be_bytes(*leaf_value.as_bytes()),
+        }
+    }
+}
+
 #[rpc(server, namespace = "admin")]
 pub(crate) trait AdminAgglayer {
+    #[method(name = "getTokenBalance")]
+    async fn get_token_balance(
+        &self,
+        network_id: NetworkId,
+        token_info: Option<TokenInfo>,
+    ) -> RpcResult<Vec<TokenBalanceEntry>>;
```

Comment:
In the interest of future extensibility, it may make sense to return a struct with a named field `balances` containing the Vec. This way, you will be able to add more fields to the response later relatively easily.